### PR TITLE
Add `CopyToClipboard` button to copy executed command in `Source` tab

### LINKED ISSUE #677: "Copy to Clipboard" button for the executed command field in check source
Current one has to select and copy the executed command from the source tab of a host or service by hand to execute the command for debugging etc.

While this is not a problem at all, I think, this is slightly unnecessary.

## Describe the solution you'd like
A "Copy to clipboard" button, would be nice.

### PR DESCRIPTION
**Pull request by raviks789**
resolves #677 

Blocked by Icinga/ipl-web#151

### CODE REVIEW

File: public/css/widget/object-inspection.less
Comment (nilmerg) [inline public/css/widget/object-inspection.less:1]: To me, this is a bit too much CSS. I think we can place most of this in ipl-web, by using some assumptions:

It's either a child: `[data-clipboard-source] > .copy-to-clipboard`
Or a direct sibling: `[data-clipboard-source] + .copy-to-clipboard`

Of course, the JS needs to attach the data attribute if it's not there, but that shouldn't be an issue.

File: library/Icingadb/Widget/Detail/EventDetail.php [hunk 1]
```diff
@@ -185,10 +188,12 @@ protected function assembleStateChangeEvent(StateHistory $stateChange)
             if (empty($stateChange->output) && empty($stateChange->long_output)) {
                 $commandOutput = new EmptyState(t('Output unavailable.'));
             } else {
-                $commandOutput = new PluginOutputContainer(
+                $commandOutput =  new PluginOutputContainer(
```
Comment (nilmerg) [inline library/Icingadb/Widget/Detail/EventDetail.php:191]: Please remove the added space

**Review by nilmerg — CHANGES_REQUESTED** [review]
Please move the behavior to Icinga Web again. Sorry for the back and forth.

File: library/Icingadb/Common/ObjectInspectionDetail.php [hunk 2]
```diff
@@ -313,6 +319,12 @@ private function createNameValueTable(array $data, array $formatters): Table
                         $value = call_user_func($formatters[$name], $value);
                     } else {
                         $value = $this->formatJson($value);
+
+                        if ($value instanceof BaseHtmlElement) {
+                            $value = CopyToClipboard::attachTo(
+                                $value->addAttributes(['id' => $name])
```
[the remark this answers is not in the document]
Comment (nilmerg) [reply library/Icingadb/Common/ObjectInspectionDetail.php]: eh, where does this come from???

**Review by nilmerg — APPROVED** [review]

**Comment by gbin2265** [comment]
Hello, 

All the service and host pages have the error after upgrading to the latest snapshot  icingadb-web-1.0.2+82.g67fc7f5-1687561409.el7.noarch.rpm | 2023-06-24 01:50

Uncaught Error: Class 'ipl\Web\Widget\CopyToClipboard' not found in /usr/share/icingaweb2/modules/icingadb/library/Icingadb/Widget/Detail/ObjectDetail.php:422

I can't find the CopyToClipboard file anywhere.

regards,
Geert

**Comment by gbin2265** [comment]
Hello, 

I found the problem!
The latest version was not installed from the icinga-php-library because the repo order is not correct.

Here's the list of commands:
yum list --showduplicates icinga-php-library


icinga-php-library.noarch                                         0.12.0-1.el7                                                             rhel.packages.icinga
icinga-php-library.noarch                                         0.12.0+3.g2ae9c91-1687554236.el7                                         rhel.packages.icinga.snapshot
icinga-php-library.noarch                                         0.12.0+3.g9c77d12-1686344659.el7                                         rhel.packages.icinga.snapshot
icinga-php-library.noarch                                         0.12.0+3.g50a5d94-1686949414.el7                                         rhel.packages.icinga.snapshot
icinga-php-library.noarch                                         0.12.0+3.g855d297-1684530225.el7                                         rhel.packages.icinga.snapshot
icinga-php-library.noarch                                         0.12.0+3.gc7e3bf3-1685739855.el7                                         rhel.packages.icinga.snapshot
icinga-php-library.noarch                                         0.12.0+3.gdf60104-1685135016.el7                                         rhel.packages.icinga.snapshot


The latest version for yum is version 0.12.0+3.gdf60104-1685135016.el7 and not version 0.12.0+3.g2ae9c91-1687554236.el7 dated 24Jun2023
If you sort , as yum does, gdf60104 is later in version as g2ae9c91.

**Pull request merged**